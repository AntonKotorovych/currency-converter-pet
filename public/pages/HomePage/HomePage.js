"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const jsx_runtime_1 = require("react/jsx-runtime");
const MainTitle_1 = __importDefault(require("components/MainTitle"));
const CurrencyRateTicker_1 = __importDefault(require("components/CurrencyRateTicker"));
const CurrencyConverter_1 = __importDefault(require("components/CurrencyConverter"));
const CurrencyRateTable_1 = __importDefault(require("components/CurrencyRateTable"));
const HomePage_module_scss_1 = __importDefault(require("./HomePage.module.scss"));
function HomePage() {
    return ((0, jsx_runtime_1.jsxs)(jsx_runtime_1.Fragment, { children: [(0, jsx_runtime_1.jsx)(MainTitle_1.default, {}), (0, jsx_runtime_1.jsx)(CurrencyRateTicker_1.default, {}), (0, jsx_runtime_1.jsxs)("div", { className: HomePage_module_scss_1.default['main-interface'], children: [(0, jsx_runtime_1.jsx)(CurrencyConverter_1.default, {}), (0, jsx_runtime_1.jsx)(CurrencyRateTable_1.default, {})] })] }));
}
exports.default = HomePage;
