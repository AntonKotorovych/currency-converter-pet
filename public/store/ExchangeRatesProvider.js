"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ExchangeRatesProvider = exports.useExchangeRates = void 0;
const jsx_runtime_1 = require("react/jsx-runtime");
const react_1 = require("react");
const NBU_API = import.meta.env.VITE_NBU_CURRENCY_EXCHANGE_API;
const ExchangeRatesContext = (0, react_1.createContext)(null);
const useExchangeRates = () => (0, react_1.useContext)(ExchangeRatesContext);
exports.useExchangeRates = useExchangeRates;
function ExchangeRatesProvider({ children }) {
    const [exchangeRates, setExchangeRates] = (0, react_1.useState)({
        response: null,
        isLoading: true,
        error: null
    });
    (0, react_1.useEffect)(() => {
        (() => __awaiter(this, void 0, void 0, function* () {
            try {
                const response = yield fetch(NBU_API);
                const data = yield response.json();
                setExchangeRates({
                    response: data,
                    isLoading: false,
                    error: null
                });
            }
            catch (error) {
                setExchangeRates({
                    response: null,
                    isLoading: false,
                    error
                });
            }
        }))();
    }, []);
    return ((0, jsx_runtime_1.jsx)(ExchangeRatesContext.Provider, { value: exchangeRates, children: children }));
}
exports.ExchangeRatesProvider = ExchangeRatesProvider;
