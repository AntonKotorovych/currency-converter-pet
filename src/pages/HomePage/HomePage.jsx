import CurrencyRateTicker from '../../components/CurrencyRateTicker';
import Card from '../../components/Card';
import MainTitle from '../../components/MainTitle';

export default function HomePage() {
  return (
    <>
      <MainTitle />
      <Card>
        <CurrencyRateTicker />
      </Card>
    </>
  );
}
