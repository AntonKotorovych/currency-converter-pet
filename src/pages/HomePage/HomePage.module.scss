.main-interface {
  width: 100%;
  margin-top: 3rem;
  height: 65vh;
  display: flex;
  align-items: center;
  justify-content: space-between;
  gap: 1.5rem;
}
