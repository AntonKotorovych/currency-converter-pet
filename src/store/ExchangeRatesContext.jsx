import { createContext, useContext, useEffect, useState } from 'react';
import { NBU_CURRENCY_EXCHANGE_API } from '../constants';

const ExchangeRatesContext = createContext(null);

export const useExchangeRates = () => useContext(ExchangeRatesContext);

export function ExchangeRatesProvider({ children }) {
  const [exchangeRates, setExchangeRates] = useState(null);

  useEffect(() => {
    const fetchExchangeRatesData = async () => {
      try {
        const response = await fetch(NBU_CURRENCY_EXCHANGE_API);
        const data = await response.json();
        setExchangeRates(data);
      } catch (error) {
        console.error(error);
      }
    };
    fetchExchangeRatesData();
  }, []);

  return (
    <ExchangeRatesContext.Provider value={exchangeRates}>
      {children}
    </ExchangeRatesContext.Provider>
  );
}
