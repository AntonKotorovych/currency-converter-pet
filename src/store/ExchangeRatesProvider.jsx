import { createContext, useContext, useEffect, useState } from 'react';

const NBU_API = import.meta.env.VITE_NBU_CURRENCY_EXCHANGE_API;

const ExchangeRatesContext = createContext(null);

export const useExchangeRates = () => useContext(ExchangeRatesContext);

export function ExchangeRatesProvider({ children }) {
  const [exchangeRates, setExchangeRates] = useState({
    response: null,
    isLoading: true,
    error: null
  });

  useEffect(() => {
    (async () => {
      try {
        const response = await fetch(NBU_API);
        const data = await response.json();
        setExchangeRates({
          response: data,
          isLoading: false,
          error: null
        });
      } catch (error) {
        setExchangeRates({
          response: null,
          isLoading: false,
          error
        });
      }
    })();
  }, []);

  return (
    <ExchangeRatesContext.Provider value={exchangeRates}>
      {children}
    </ExchangeRatesContext.Provider>
  );
}
