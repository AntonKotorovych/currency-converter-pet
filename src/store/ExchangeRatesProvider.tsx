import {
  ReactNode,
  createContext,
  useContext,
  useEffect,
  useState
} from 'react';

import { Currency } from 'types/interfaces/index';

const NBU_API = import.meta.env.VITE_NBU_CURRENCY_EXCHANGE_API;

interface ExchangeRates {
  response: Currency[] | null;
  isLoading: boolean;
  error: { name: string; message: string } | null | unknown;
}

const ExchangeRatesContext = createContext<ExchangeRates | null>(null);

export const useExchangeRates = () =>
  useContext(ExchangeRatesContext) as ExchangeRates;

export function ExchangeRatesProvider({ children }: { children: ReactNode }) {
  const [exchangeRates, setExchangeRates] = useState<ExchangeRates>({
    response: null,
    isLoading: true,
    error: null
  });

  useEffect(() => {
    (async () => {
      try {
        const response = await fetch(NBU_API);
        const data = await response.json();
        setExchangeRates({
          response: data,
          isLoading: false,
          error: null
        });
      } catch (error) {
        setExchangeRates({
          response: null,
          isLoading: false,
          error
        });
      }
    })();
  }, []);

  return (
    <ExchangeRatesContext.Provider value={exchangeRates}>
      {children}
    </ExchangeRatesContext.Provider>
  );
}
