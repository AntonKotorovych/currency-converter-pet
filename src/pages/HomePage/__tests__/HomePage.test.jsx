import { screen, render } from '@testing-library/react';
import { mockResponse } from 'mocks/exchangeRatesResponse';
import { useExchangeRates } from 'store/ExchangeRatesProvider';
import HomePage from '..';

jest.mock('store/ExchangeRatesProvider');

describe('HomePage', () => {
  const defaultRatesState = {
    response: mockResponse
  };

  const setExchangeRates = (state = defaultRatesState) =>
    useExchangeRates.mockReturnValue(state);

  const renderComponent = () => render(<HomePage />);

  beforeEach(() => {
    jest.clearAllMocks();
    setExchangeRates();
  });

  beforeAll(() => {
    jest.useFakeTimers();
    jest.setSystemTime(new Date(2022, 1, 24));
  });

  afterAll(() => {
    jest.useRealTimers();
  });

  test('matches snapshot', () => {
    const { container } = renderComponent();

    screen.debug();

    expect(container).toMatchSnapshot();
  });
});
