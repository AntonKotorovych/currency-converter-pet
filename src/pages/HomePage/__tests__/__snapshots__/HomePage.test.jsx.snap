// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage matches snapshot 1`] = `
<div>
  <div
    class="main-title-container"
  >
    <h1>
      Актуальні Курси Валют станом на 
      24.02.2022
    </h1>
  </div>
  <div
    class="currency-rate-container"
  >
    <ul
      class="currency-rate-list"
    >
      <li
        class="list-item"
      >
        Австралійський долар
         - 
        25.8124
         ₴
      </li>
      <li
        class="list-item"
      >
        Канадський долар
         - 
        28.721
         ₴
      </li>
      <li
        class="list-item"
      >
        Юань Женьміньбі
         - 
        5.391
         ₴
      </li>
      <li
        class="list-item"
      >
        Чеська крона
         - 
        1.6689
         ₴
      </li>
      <li
        class="list-item"
      >
        Данська крона
         - 
        5.6781
         ₴
      </li>
      <li
        class="list-item"
      >
        Гонконгівський долар
         - 
        4.9797
         ₴
      </li>
      <li
        class="list-item"
      >
        Форинт
         - 
        0.108859
         ₴
      </li>
      <li
        class="list-item"
      >
        Індійська рупія
         - 
        0.46801
         ₴
      </li>
      <li
        class="list-item"
      >
        Рупія
         - 
        0.0024573
         ₴
      </li>
      <li
        class="list-item"
      >
        Новий ізраїльський шекель
         - 
        10.5607
         ₴
      </li>
      <li
        class="list-item"
      >
        Єна
         - 
        0.25683
         ₴
      </li>
      <li
        class="list-item"
      >
        Теньге
         - 
        0.087285
         ₴
      </li>
      <li
        class="list-item"
      >
        Вона
         - 
        0.028779
         ₴
      </li>
      <li
        class="list-item"
      >
        Мексиканське песо
         - 
        2.3908
         ₴
      </li>
      <li
        class="list-item"
      >
        Молдовський лей
         - 
        2.2031
         ₴
      </li>
      <li
        class="list-item"
      >
        Новозеландський долар
         - 
        23.6073
         ₴
      </li>
      <li
        class="list-item"
      >
        Норвезька крона
         - 
        3.6529
         ₴
      </li>
      <li
        class="list-item"
      >
        Російський рубль
         - 
        0.42009
         ₴
      </li>
      <li
        class="list-item"
      >
        Сінгапурський долар
         - 
        28.9707
         ₴
      </li>
      <li
        class="list-item"
      >
        Ренд
         - 
        2.1066
         ₴
      </li>
      <li
        class="list-item"
      >
        Шведська крона
         - 
        3.6968
         ₴
      </li>
      <li
        class="list-item"
      >
        Швейцарський франк
         - 
        43.1427
         ₴
      </li>
      <li
        class="list-item"
      >
        Єгипетський фунт
         - 
        0.8201
         ₴
      </li>
      <li
        class="list-item"
      >
        Фунт стерлінгів
         - 
        49.4411
         ₴
      </li>
      <li
        class="list-item"
      >
        Долар США
         - 
        38.9945
         ₴
      </li>
      <li
        class="list-item"
      >
        Білоруський рубль
         - 
        14.1736
         ₴
      </li>
      <li
        class="list-item"
      >
        Азербайджанський манат
         - 
        22.942
         ₴
      </li>
      <li
        class="list-item"
      >
        Румунський лей
         - 
        8.5247
         ₴
      </li>
      <li
        class="list-item"
      >
        Турецька ліра
         - 
        1.2113
         ₴
      </li>
      <li
        class="list-item"
      >
        СПЗ (спеціальні права запозичення)
         - 
        51.7542
         ₴
      </li>
      <li
        class="list-item"
      >
        Болгарський лев
         - 
        21.6552
         ₴
      </li>
      <li
        class="list-item"
      >
        Євро
         - 
        42.3519
         ₴
      </li>
      <li
        class="list-item"
      >
        Злотий
         - 
        9.9519
         ₴
      </li>
      <li
        class="list-item"
      >
        Алжирський динар
         - 
        0.29133
         ₴
      </li>
      <li
        class="list-item"
      >
        Така
         - 
        0.35452
         ₴
      </li>
      <li
        class="list-item"
      >
        Вірменський драм
         - 
        0.099159
         ₴
      </li>
      <li
        class="list-item"
      >
        Домініканське песо
         - 
        0.66115
         ₴
      </li>
      <li
        class="list-item"
      >
        Іранський ріал
         - 
        0.00092851
         ₴
      </li>
      <li
        class="list-item"
      >
        Іракський динар
         - 
        0.029769
         ₴
      </li>
      <li
        class="list-item"
      >
        Сом
         - 
        0.43587
         ₴
      </li>
      <li
        class="list-item"
      >
        Ліванський фунт
         - 
        0.000436
         ₴
      </li>
      <li
        class="list-item"
      >
        Лівійський динар
         - 
        8.0586
         ₴
      </li>
      <li
        class="list-item"
      >
        Малайзійський ринггіт
         - 
        8.2657
         ₴
      </li>
      <li
        class="list-item"
      >
        Марокканський дирхам
         - 
        3.8465
         ₴
      </li>
      <li
        class="list-item"
      >
        Пакистанська рупія
         - 
        0.14042
         ₴
      </li>
      <li
        class="list-item"
      >
        Саудівський ріял
         - 
        10.3982
         ₴
      </li>
      <li
        class="list-item"
      >
        Донг
         - 
        0.0015718
         ₴
      </li>
      <li
        class="list-item"
      >
        Бат
         - 
        1.07729
         ₴
      </li>
      <li
        class="list-item"
      >
        Дирхам ОАЕ
         - 
        10.6188
         ₴
      </li>
      <li
        class="list-item"
      >
        Туніський динар
         - 
        12.4776
         ₴
      </li>
      <li
        class="list-item"
      >
        Узбецький сум
         - 
        0.0030882
         ₴
      </li>
      <li
        class="list-item"
      >
        Новий тайванський долар
         - 
        1.221
         ₴
      </li>
      <li
        class="list-item"
      >
        Туркменський новий манат
         - 
        11.1421
         ₴
      </li>
      <li
        class="list-item"
      >
        Сербський динар
         - 
        0.36025
         ₴
      </li>
      <li
        class="list-item"
      >
        Сомоні
         - 
        3.561
         ₴
      </li>
      <li
        class="list-item"
      >
        Ларі
         - 
        14.9113
         ₴
      </li>
      <li
        class="list-item"
      >
        Бразильський реал
         - 
        7.2282
         ₴
      </li>
      <li
        class="list-item"
      >
        Золото
         - 
        91816.06
         ₴
      </li>
      <li
        class="list-item"
      >
        Срібло
         - 
        1094.07
         ₴
      </li>
      <li
        class="list-item"
      >
        Платина
         - 
        38216.95
         ₴
      </li>
      <li
        class="list-item"
      >
        Паладій
         - 
        41443.74
         ₴
      </li>
    </ul>
  </div>
  <div
    class="main-interface"
  >
    <div
      class="currency-converter"
    >
      <div
        class="currency-converter__block-main"
      >
        <div>
          <input
            data-testid="firstInput"
            id="firstInput"
            name="firstInput"
            step="0.01"
            type="number"
            value="38.99"
          />
        </div>
        <label
          for="firstInput"
        >
          UAH
        </label>
      </div>
      <div
        class="currency-converter__block-secondary"
      >
        <div>
          <input
            data-testid="secondInput"
            id="secondInput"
            name="secondInput"
            step="0.01"
            type="number"
            value="1"
          />
        </div>
        <div
          class="input-container"
        >
          <select>
            <option
              value="AUD"
            >
              Австралійський долар
            </option>
            <option
              value="CAD"
            >
              Канадський долар
            </option>
            <option
              value="CNY"
            >
              Юань Женьміньбі
            </option>
            <option
              value="CZK"
            >
              Чеська крона
            </option>
            <option
              value="DKK"
            >
              Данська крона
            </option>
            <option
              value="HKD"
            >
              Гонконгівський долар
            </option>
            <option
              value="HUF"
            >
              Форинт
            </option>
            <option
              value="INR"
            >
              Індійська рупія
            </option>
            <option
              value="IDR"
            >
              Рупія
            </option>
            <option
              value="ILS"
            >
              Новий ізраїльський шекель
            </option>
            <option
              value="JPY"
            >
              Єна
            </option>
            <option
              value="KZT"
            >
              Теньге
            </option>
            <option
              value="KRW"
            >
              Вона
            </option>
            <option
              value="MXN"
            >
              Мексиканське песо
            </option>
            <option
              value="MDL"
            >
              Молдовський лей
            </option>
            <option
              value="NZD"
            >
              Новозеландський долар
            </option>
            <option
              value="NOK"
            >
              Норвезька крона
            </option>
            <option
              value="RUB"
            >
              Російський рубль
            </option>
            <option
              value="SGD"
            >
              Сінгапурський долар
            </option>
            <option
              value="ZAR"
            >
              Ренд
            </option>
            <option
              value="SEK"
            >
              Шведська крона
            </option>
            <option
              value="CHF"
            >
              Швейцарський франк
            </option>
            <option
              value="EGP"
            >
              Єгипетський фунт
            </option>
            <option
              value="GBP"
            >
              Фунт стерлінгів
            </option>
            <option
              value="USD"
            >
              Долар США
            </option>
            <option
              value="BYN"
            >
              Білоруський рубль
            </option>
            <option
              value="AZN"
            >
              Азербайджанський манат
            </option>
            <option
              value="RON"
            >
              Румунський лей
            </option>
            <option
              value="TRY"
            >
              Турецька ліра
            </option>
            <option
              value="XDR"
            >
              СПЗ (спеціальні права запозичення)
            </option>
            <option
              value="BGN"
            >
              Болгарський лев
            </option>
            <option
              value="EUR"
            >
              Євро
            </option>
            <option
              value="PLN"
            >
              Злотий
            </option>
            <option
              value="DZD"
            >
              Алжирський динар
            </option>
            <option
              value="BDT"
            >
              Така
            </option>
            <option
              value="AMD"
            >
              Вірменський драм
            </option>
            <option
              value="DOP"
            >
              Домініканське песо
            </option>
            <option
              value="IRR"
            >
              Іранський ріал
            </option>
            <option
              value="IQD"
            >
              Іракський динар
            </option>
            <option
              value="KGS"
            >
              Сом
            </option>
            <option
              value="LBP"
            >
              Ліванський фунт
            </option>
            <option
              value="LYD"
            >
              Лівійський динар
            </option>
            <option
              value="MYR"
            >
              Малайзійський ринггіт
            </option>
            <option
              value="MAD"
            >
              Марокканський дирхам
            </option>
            <option
              value="PKR"
            >
              Пакистанська рупія
            </option>
            <option
              value="SAR"
            >
              Саудівський ріял
            </option>
            <option
              value="VND"
            >
              Донг
            </option>
            <option
              value="THB"
            >
              Бат
            </option>
            <option
              value="AED"
            >
              Дирхам ОАЕ
            </option>
            <option
              value="TND"
            >
              Туніський динар
            </option>
            <option
              value="UZS"
            >
              Узбецький сум
            </option>
            <option
              value="TWD"
            >
              Новий тайванський долар
            </option>
            <option
              value="TMT"
            >
              Туркменський новий манат
            </option>
            <option
              value="RSD"
            >
              Сербський динар
            </option>
            <option
              value="TJS"
            >
              Сомоні
            </option>
            <option
              value="GEL"
            >
              Ларі
            </option>
            <option
              value="BRL"
            >
              Бразильський реал
            </option>
            <option
              value="XAU"
            >
              Золото
            </option>
            <option
              value="XAG"
            >
              Срібло
            </option>
            <option
              value="XPT"
            >
              Платина
            </option>
            <option
              value="XPD"
            >
              Паладій
            </option>
          </select>
        </div>
      </div>
    </div>
    <div
      class="table-wrapper"
    >
      <div
        class="currency-search"
      >
        <div>
          <input
            data-testid="searchInput"
            name="search"
            placeholder="Пошук Валюти..."
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class="currency-rate-table-container"
      >
        <table
          class="currency-rate-table"
        >
          <thead
            class="thead"
          >
            <tr
              class="currency-rate-table__head"
            >
              <th>
                Валюта
              </th>
              <th>
                Курс
              </th>
            </tr>
          </thead>
          <tbody>
            <tr
              class="currency-rate-table__row"
            >
              <td>
                Австралійський долар
              </td>
              <td>
                25.8124
              </td>
            </tr>
            <tr
              class="currency-rate-table__row"
            >
              <td>
                Канадський долар
              </td>
              <td>
                28.721
              </td>
            </tr>
            <tr
              class="currency-rate-table__row"
            >
              <td>
                Юань Женьміньбі
              </td>
              <td>
                5.391
              </td>
            </tr>
            <tr
              class="currency-rate-table__row"
            >
              <td>
                Чеська крона
              </td>
              <td>
                1.6689
              </td>
            </tr>
            <tr
              class="currency-rate-table__row"
            >
              <td>
                Данська крона
              </td>
              <td>
                5.6781
              </td>
            </tr>
            <tr
              class="currency-rate-table__row"
            >
              <td>
                Гонконгівський долар
              </td>
              <td>
                4.9797
              </td>
            </tr>
            <tr
              class="currency-rate-table__row"
            >
              <td>
                Форинт
              </td>
              <td>
                0.108859
              </td>
            </tr>
            <tr
              class="currency-rate-table__row"
            >
              <td>
                Індійська рупія
              </td>
              <td>
                0.46801
              </td>
            </tr>
            <tr
              class="currency-rate-table__row"
            >
              <td>
                Рупія
              </td>
              <td>
                0.0024573
              </td>
            </tr>
            <tr
              class="currency-rate-table__row"
            >
              <td>
                Новий ізраїльський шекель
              </td>
              <td>
                10.5607
              </td>
            </tr>
            <tr
              class="currency-rate-table__row"
            >
              <td>
                Єна
              </td>
              <td>
                0.25683
              </td>
            </tr>
            <tr
              class="currency-rate-table__row"
            >
              <td>
                Теньге
              </td>
              <td>
                0.087285
              </td>
            </tr>
            <tr
              class="currency-rate-table__row"
            >
              <td>
                Вона
              </td>
              <td>
                0.028779
              </td>
            </tr>
            <tr
              class="currency-rate-table__row"
            >
              <td>
                Мексиканське песо
              </td>
              <td>
                2.3908
              </td>
            </tr>
            <tr
              class="currency-rate-table__row"
            >
              <td>
                Молдовський лей
              </td>
              <td>
                2.2031
              </td>
            </tr>
            <tr
              class="currency-rate-table__row"
            >
              <td>
                Новозеландський долар
              </td>
              <td>
                23.6073
              </td>
            </tr>
            <tr
              class="currency-rate-table__row"
            >
              <td>
                Норвезька крона
              </td>
              <td>
                3.6529
              </td>
            </tr>
            <tr
              class="currency-rate-table__row"
            >
              <td>
                Російський рубль
              </td>
              <td>
                0.42009
              </td>
            </tr>
            <tr
              class="currency-rate-table__row"
            >
              <td>
                Сінгапурський долар
              </td>
              <td>
                28.9707
              </td>
            </tr>
            <tr
              class="currency-rate-table__row"
            >
              <td>
                Ренд
              </td>
              <td>
                2.1066
              </td>
            </tr>
            <tr
              class="currency-rate-table__row"
            >
              <td>
                Шведська крона
              </td>
              <td>
                3.6968
              </td>
            </tr>
            <tr
              class="currency-rate-table__row"
            >
              <td>
                Швейцарський франк
              </td>
              <td>
                43.1427
              </td>
            </tr>
            <tr
              class="currency-rate-table__row"
            >
              <td>
                Єгипетський фунт
              </td>
              <td>
                0.8201
              </td>
            </tr>
            <tr
              class="currency-rate-table__row"
            >
              <td>
                Фунт стерлінгів
              </td>
              <td>
                49.4411
              </td>
            </tr>
            <tr
              class="currency-rate-table__row"
            >
              <td>
                Долар США
              </td>
              <td>
                38.9945
              </td>
            </tr>
            <tr
              class="currency-rate-table__row"
            >
              <td>
                Білоруський рубль
              </td>
              <td>
                14.1736
              </td>
            </tr>
            <tr
              class="currency-rate-table__row"
            >
              <td>
                Азербайджанський манат
              </td>
              <td>
                22.942
              </td>
            </tr>
            <tr
              class="currency-rate-table__row"
            >
              <td>
                Румунський лей
              </td>
              <td>
                8.5247
              </td>
            </tr>
            <tr
              class="currency-rate-table__row"
            >
              <td>
                Турецька ліра
              </td>
              <td>
                1.2113
              </td>
            </tr>
            <tr
              class="currency-rate-table__row"
            >
              <td>
                СПЗ (спеціальні права запозичення)
              </td>
              <td>
                51.7542
              </td>
            </tr>
            <tr
              class="currency-rate-table__row"
            >
              <td>
                Болгарський лев
              </td>
              <td>
                21.6552
              </td>
            </tr>
            <tr
              class="currency-rate-table__row"
            >
              <td>
                Євро
              </td>
              <td>
                42.3519
              </td>
            </tr>
            <tr
              class="currency-rate-table__row"
            >
              <td>
                Злотий
              </td>
              <td>
                9.9519
              </td>
            </tr>
            <tr
              class="currency-rate-table__row"
            >
              <td>
                Алжирський динар
              </td>
              <td>
                0.29133
              </td>
            </tr>
            <tr
              class="currency-rate-table__row"
            >
              <td>
                Така
              </td>
              <td>
                0.35452
              </td>
            </tr>
            <tr
              class="currency-rate-table__row"
            >
              <td>
                Вірменський драм
              </td>
              <td>
                0.099159
              </td>
            </tr>
            <tr
              class="currency-rate-table__row"
            >
              <td>
                Домініканське песо
              </td>
              <td>
                0.66115
              </td>
            </tr>
            <tr
              class="currency-rate-table__row"
            >
              <td>
                Іранський ріал
              </td>
              <td>
                0.00092851
              </td>
            </tr>
            <tr
              class="currency-rate-table__row"
            >
              <td>
                Іракський динар
              </td>
              <td>
                0.029769
              </td>
            </tr>
            <tr
              class="currency-rate-table__row"
            >
              <td>
                Сом
              </td>
              <td>
                0.43587
              </td>
            </tr>
            <tr
              class="currency-rate-table__row"
            >
              <td>
                Ліванський фунт
              </td>
              <td>
                0.000436
              </td>
            </tr>
            <tr
              class="currency-rate-table__row"
            >
              <td>
                Лівійський динар
              </td>
              <td>
                8.0586
              </td>
            </tr>
            <tr
              class="currency-rate-table__row"
            >
              <td>
                Малайзійський ринггіт
              </td>
              <td>
                8.2657
              </td>
            </tr>
            <tr
              class="currency-rate-table__row"
            >
              <td>
                Марокканський дирхам
              </td>
              <td>
                3.8465
              </td>
            </tr>
            <tr
              class="currency-rate-table__row"
            >
              <td>
                Пакистанська рупія
              </td>
              <td>
                0.14042
              </td>
            </tr>
            <tr
              class="currency-rate-table__row"
            >
              <td>
                Саудівський ріял
              </td>
              <td>
                10.3982
              </td>
            </tr>
            <tr
              class="currency-rate-table__row"
            >
              <td>
                Донг
              </td>
              <td>
                0.0015718
              </td>
            </tr>
            <tr
              class="currency-rate-table__row"
            >
              <td>
                Бат
              </td>
              <td>
                1.07729
              </td>
            </tr>
            <tr
              class="currency-rate-table__row"
            >
              <td>
                Дирхам ОАЕ
              </td>
              <td>
                10.6188
              </td>
            </tr>
            <tr
              class="currency-rate-table__row"
            >
              <td>
                Туніський динар
              </td>
              <td>
                12.4776
              </td>
            </tr>
            <tr
              class="currency-rate-table__row"
            >
              <td>
                Узбецький сум
              </td>
              <td>
                0.0030882
              </td>
            </tr>
            <tr
              class="currency-rate-table__row"
            >
              <td>
                Новий тайванський долар
              </td>
              <td>
                1.221
              </td>
            </tr>
            <tr
              class="currency-rate-table__row"
            >
              <td>
                Туркменський новий манат
              </td>
              <td>
                11.1421
              </td>
            </tr>
            <tr
              class="currency-rate-table__row"
            >
              <td>
                Сербський динар
              </td>
              <td>
                0.36025
              </td>
            </tr>
            <tr
              class="currency-rate-table__row"
            >
              <td>
                Сомоні
              </td>
              <td>
                3.561
              </td>
            </tr>
            <tr
              class="currency-rate-table__row"
            >
              <td>
                Ларі
              </td>
              <td>
                14.9113
              </td>
            </tr>
            <tr
              class="currency-rate-table__row"
            >
              <td>
                Бразильський реал
              </td>
              <td>
                7.2282
              </td>
            </tr>
            <tr
              class="currency-rate-table__row"
            >
              <td>
                Золото
              </td>
              <td>
                91816.06
              </td>
            </tr>
            <tr
              class="currency-rate-table__row"
            >
              <td>
                Срібло
              </td>
              <td>
                1094.07
              </td>
            </tr>
            <tr
              class="currency-rate-table__row"
            >
              <td>
                Платина
              </td>
              <td>
                38216.95
              </td>
            </tr>
            <tr
              class="currency-rate-table__row"
            >
              <td>
                Паладій
              </td>
              <td>
                41443.74
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</div>
`;
